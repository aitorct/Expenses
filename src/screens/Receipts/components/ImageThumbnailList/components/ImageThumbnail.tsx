import React from 'react';
import {StyleSheet, TouchableOpacity, View} from 'react-native';
import FastImage from 'react-native-fast-image';
import colors from '@theme/colors';
import sizes from '@theme/sizes';

interface Props {
  onSelect: () => void;
  isSelected: boolean;
  imageUri: string;
}

const ImageThumbnail = ({onSelect, isSelected, imageUri}: Props) => {
  return (
    <View>
      {isSelected && (
        <View style={styles.indicatorWrapper}>
          <View style={styles.indicator} />
        </View>
      )}
      <TouchableOpacity onPress={onSelect}>
        <FastImage source={{uri: imageUri}} style={styles.image} />
      </TouchableOpacity>
    </View>
  );
};

const styles = StyleSheet.create({
  indicatorWrapper: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'center',
    paddingBottom: sizes.s8,
  },
  indicator: {
    height: sizes.dotIndicator,
    width: sizes.dotIndicator,
    borderRadius: sizes.dotIndicator / 2,
    backgroundColor: colors.tertiaryBackground,
  },
  image: {
    width: sizes.square,
    height: sizes.square,
  },
});

export default ImageThumbnail;
