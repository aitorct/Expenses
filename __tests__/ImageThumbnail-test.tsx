import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import ImageThumbnail from '@screens/Receipts/components/ImageThumbnailList/components/ImageThumbnail';

const mockedFunction = jest.fn(() => {});

const imageUri = 'file://pictures/image.jpg';

it('Unselected ImageThumbnail renders correctly', () => {
  const tree = renderer
    .create(
      <ImageThumbnail
        onSelect={mockedFunction}
        isSelected={false}
        imageUri={imageUri}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('Selected ImageThumbnail renders correctly', () => {
  const tree = renderer
    .create(
      <ImageThumbnail
        onSelect={mockedFunction}
        isSelected={true}
        imageUri={imageUri}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
