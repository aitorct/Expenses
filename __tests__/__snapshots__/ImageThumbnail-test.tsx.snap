// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Selected ImageThumbnail renders correctly 1`] = `
<View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
        "paddingBottom": 8,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#000000",
          "borderRadius": 5,
          "height": 10,
          "width": 10,
        }
      }
    />
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Object {
            "height": 100,
            "width": 100,
          },
        ]
      }
    >
      <FastImageView
        resizeMode="cover"
        source={
          Object {
            "uri": "file://pictures/image.jpg",
          }
        }
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      />
    </View>
  </View>
</View>
`;

exports[`Unselected ImageThumbnail renders correctly 1`] = `
<View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Object {
            "height": 100,
            "width": 100,
          },
        ]
      }
    >
      <FastImageView
        resizeMode="cover"
        source={
          Object {
            "uri": "file://pictures/image.jpg",
          }
        }
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      />
    </View>
  </View>
</View>
`;
